prusti-encoder/src/encoders/ty/indirect.rs

```diff
@@ -74,18 +73,37 @@ impl TaskEncoder for IndirectPredicatesEnc {
         task_key: &Self::TaskKey<'vir>,
         deps: &mut TaskEncoderDependencies<'vir, Self>,
     ) -> EncodeFullResult<'vir, Self> {
+        deps.emit_output_ref(
+            *task_key,
+            (),
+        )?;
         vir::with_vcx(|vcx| {
             let (ty, proj_region) = task_key;
-            let self_ty_enc = deps.require_local::<TyPureEnc>(*ty)?;
+            let self_ty_enc = deps.require_dep::<TyUsePureEnc>(*ty)?;
             let mut covariant = Vec::<ExprOutput<'vir>>::new();
             let mut contravariant = Vec::<ExprOutput<'vir>>::new();
-            match ty.kind() {
-                ty::TyKind::Ref(ref_region, inner_ty, ty::Mutability::Mut) => {
-                    let ref_domain = self_ty_enc.expect_mutref();
-                    if IndirectKey::from_region(*ref_region)
+            let combined = ty.ty.zip(self_ty_enc);
+            match combined.specifics {
+                // Optimisation: if there are no type arguments, there cannot be
+                // anything behind a ref inside (except for 'static, which we
+                // ignore for now).
```

Comment:
I guess this is true, but does it save much in terms of performance? (And do we care?)

Reply:
It avoids unsupported types :)

Reply:
Ok, fair, but then please say so in the comment, otherwise I would assume the "optimisation" branch can freely be removed if needed.